### PR TITLE
Allow PactGroup.wait to be eager (polling all pacts)

### PR DESCRIPTION
**Pull request by ayalash**

### CODE REVIEW

File: pact/group.py [hunk 1]
```diff
@@ -26,20 +26,23 @@ def add(self, pact, absorb=False):
...
             while pact._then:
                 # then might throw, so we attempt it first
                 self.then(pact._then[0])
                 pact._then.pop(0)
 
     def _is_finished(self):
-        while self._pacts:
-            pact = self._pacts[0]
+        has_finished = True
+        for pact in self._pacts[:]:
```
Comment (vmalloc) [inline pact/group.py]: Why do we copy the list here? It's a bit redundant...

**Comment by ayalash** [comment]
@vmalloc I pushed a new commit that fixes all your review's comments.
I'm not sure about my documentation addition, please review it as well.

File: doc/user_guide.rst [hunk 2]
```diff
@@ -104,6 +104,25 @@ And of course it will be more descriptive when only one pact was not satisfied:
...
+Waiting for group is a lazy operation, by default, which means that will poll pacts only if previous pact had finished:
+
+.. code-block:: python
+
+       >>> pact_a = pact_delete_async('/path_a').during(print, 'a', end='').then(print, 'A', end='')
+       >>> pact_b = pact_delete_async('/path_b').during(print, 'b', end='').then(print, 'B')
+       >>> PactGroup([pact_a, pact_b]).wait()
+       aaaaaaaaaaaAbB
+
+Group can be poll eagerly by passing ``lazy=True`` to its creation:
```
Comment (vmalloc) [inline doc/user_guide.rst]: lazy=False. I will fix it.

**Comment by vmalloc** [comment]
Thanks!

**Pull request merged**